src/Select.js

```diff
@@ -879,9 +880,17 @@ const Select = React.createClass({
 		}
 
 		for (var i = 0; i < options.length; i++) {
-			if (!options[i].disabled) return i;
+			// if there is no selected option just use first value in options
+			if (!options[i].disabled && !focusedOption) {
+				return i;
+			} else if (!options[i].disabled && options[i].value === focusedOption.value) {
```

Comment:
Please use the reference to the `this.props.valueKey` instead of the `.value`.

(Sorry for the previous noise with the comments, I was comparing it against the 1.0.0.beta13 build and not the master.